Fix NPM publication by adding build step

tylerthebuildor opened the pull request:
The current NPM package is broken, because a projects local version of Babel does not transpile node_modules automatically. We can fix this by automatically transpiling to es5 before publishing to NPM.

You don't have to do anything different just pull down the latest changes, npm install, run ./publish again, and choose patch.

```bash
git pull origin master
npm install
./publish
```

tylerthebuildor committed:
Merge branch 'add-npm'

tylerthebuildor committed:
Add build step

tylerthebuildor committed:
Merge branch 'master' of github.com:tylerbuchea/react-material-ui-rating

tylerthebuildor committed:
Ignore index.js

tylerthebuildor commented:
Any word on this? Would love to get it published 😄

lawrentiy commented on Rating.js:
```diff
@@ -1,6 +1,4 @@
 import React from 'react';
-import Star from 'material-ui/svg-icons/toggle/star'
-import StarBorder from 'material-ui/svg-icons/toggle/star-border'
 import { StarBorder, Star } from 'material-ui-icons';
```
That have to remove 
`import { StarBorder, Star } from 'material-ui-icons';`
instead
`
import Star from 'material-ui/svg-icons/toggle/star'
import StarBorder from 'material-ui/svg-icons/toggle/star-border'
`
. See #1

lawrentiy commented:
Please, see my code review

tylerthebuildor committed:
Add peerDeps section to package.json and update README to call this out

tylerthebuildor commented:
So those are the old requires pre v1.0 my thought is if people want to add this library to their pre 1.0 material ui project. They can just `npm install react-material-ui-rating@old-version`

That way we don't have to keep supporting past versions. This is sort of the standard way of handling past projects. Because material-ui will change again and then we'd have to support three versions. You'll see what I mean if you install the latest stuff:

```bash
npm install --save material-ui@next
npm install --save material-ui-icons
```

I've added a new commit that calls this versioning thing out in the README and I also added a "peerDependcies" section in the package.json so there isn't any confusion.

I'm open to suggestions though what do you think?

tylerthebuildor commented:
Does that make sense? I think we're super close to having the killer rating component here 😄  Great work!
